gateway-js/src/__tests__/buildQueryPlan.test.ts

```diff
@@ -1171,4 +1173,31 @@ describe('buildQueryPlan', () => {
       `);
     });
   });
+
+  describe('overridden fields and type', () => {
+    it(`query plan of overridden field`, () => {
```

Comment:
Fwiw, I dislike the test fixtures/the tests using them. I find those tests annoying to work with because you need to navigate a bunch of files to even start to figure out what the test even does (unless you happen to know the fixtures intimately, and well, I don't). Not to mention that when tons of tests use the same fixtures, it's much too easy to modify them for some new test without realising that the changes invalidates some existing test (by which I mean, those existing test might still pass but stop actually testing what they were meant to).

Anyway, that was my rant, and it's ok to not agree (and totally fine keeping that test for this PR), but if you simply happened to have stumbled on that test file and though this was the status quo, I'll note that the query planning tests I've personally added have been added mostly to `query-planner-js/src/__tests__/buildPlan.test.ts` and are self-contained test.

But that point aside, I think this test would be more useful if the `@override` was in `accounts` rather than `books`. Because as it stand, I'd expect the query plan of this test _even_ if `@override` wasn't working properly because that's obviously more efficient that going into `accounts`. But having the `@override` in `accounts` would allow to validate that the query planner _has to_ force a jump into `accounts` and cannot just use the definition of `description` in `books`.

Reply:
I'm not going to change out of the using the fixtures now, but I will in the future as I agree that it makes things less straightforward.